src/Sentry.DiagnosticSource/Internal/DiagnosticSource/EFDiagnosticSourceHelper.cs

```diff
@@ -0,0 +1,117 @@
+    }
+
+    internal void AddSpan()
+    {
+        Options.LogDebug($"(Sentry add span {Operation})");
+        LogTransactionSpans();
+        if (Transaction == null)
+        {
+            return;
+        }
+
+        var parent = GetParentSpan(Transaction);
+        var child = parent.StartChild(Operation, Description);
+
+        SetSpanReference(child);
+    }
+
+    internal void FinishSpan(SpanStatus status)
+    {
+        if (Transaction == null)
+        {
+            return;
+        }
+
+        Options.LogDebug($"(Sentry finish span {Operation})");
+        LogTransactionSpans();
+
+        var sourceSpan = GetSpanReference(Transaction);
+        if (sourceSpan == null)
+        {
+            Options.LogWarning("Trying to close a span that was already garbage collected. {0}", Operation);
+            return;
+        }
+
+        sourceSpan.Finish(status);
+    }
+
+    private void LogTransactionSpans()
+    {
+        if (Transaction == null)
+        {
+            Options.LogDebug($"(Sentry transaction is null)");
+            return;
+        }
+
+        Options.LogDebug("Transaction Spans");
+        Options.LogDebug("-----------------");
+        foreach (var span in Transaction.Spans)
+        {
+            Options.LogDebug($"id: {span.SpanId} operation: {span.Operation}");
+        }
+    }
+
+    /// <summary>
+    /// Get the Query with error message and remove the unneeded values.
+    /// </summary>
+    /// <example>
+    /// Compiling query model:
+    /// EF initialize...\r\nEF Query...
+    /// becomes:
+    /// EF Query...
+    /// </example>
+    /// <param name="value">the query to be parsed value</param>
+    /// <returns>the filtered query</returns>
+    internal static string? FilterNewLineValue(object? value)
+    {
+        var str = value?.ToString();
+        return str?[(str.IndexOf('\n') + 1)..];
+    }
+
+    /// <summary>
+    /// Finds an appropriate parent for EF diagnostic spans. Note that in our implementaion we're "flattening" these.
+    /// Spans from <see cref="EFQueryCompilerDiagnosticSourceHelper"/> and <see cref="EFCommandDiagnosticSourceHelper"/>
+    /// will both have the same parent as those of <see cref="EFConnectionDiagnosticSourceHelper"/>.
+    ///
+    /// We've done this for two reasons:
+    /// 1. We could show these underneath the relevant connection, but each command often gets it's own connection which
+    ///    makes the resulting waterfall diagram hard to read.
+    /// 2. Sentry has a performance to errors feature which detects n + 1 problems on OM frameworks... but this only
+    ///    works if all of the "n" queries have the same parent span in the transaction. Again, since queries ofen run
+    ///    on their own connection, if we wrap each query in a connection span, it breaks the performance error handling
+    /// </summary>
+    private ISpan GetParentSpan(ITransaction transaction) => transaction.GetDbParentSpan();
```

Comment:
Other than a place to put the comment, I'm not sure this function is adding anything useful.  If we need to keep it, then it can simplify to:

```suggestion
    private static ISpan GetParentSpan(ISpan span) => span.GetDbParentSpan();
```

But I think it could just be inlined.  There's only one usage.